pkg/util/kubernetes/hpa/hpa_test.go

```diff
@@ -0,0 +1,157 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2017 Datadog, Inc.
+
+// +build kubeapiserver
+
+package hpa
+
+import (
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+
+	"encoding/json"
+	"fmt"
+
+	"github.com/stretchr/testify/assert"
+	"k8s.io/api/autoscaling/v2beta1"
+	"k8s.io/api/core/v1"
+)
+
+func newMockConfigMap(metricName string, labels map[string]string) *v1.ConfigMap {
+	cm := &v1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			Kind: "ConfigMap",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      datadogHPAConfigMap,
+			Namespace: "default",
+		},
+	}
+	if metricName == "" || len(labels) == 0 {
+		return cm
+	}
+
+	custMetric := CustomExternalMetric{
+		Name:      metricName,
+		Labels:    labels,
+		Timestamp: 12,
+		HpaName:   "foo",
+		Value:     1,
+		Valid:     false,
+	}
+	cm.Data = make(map[string]string)
+	marsh, _ := json.Marshal(custMetric)
+	cm.Data[metricName] = string(marsh)
+	return cm
+}
+
+func newMockHPAExternalManifest(metricName string, labels map[string]string) *v2beta1.HorizontalPodAutoscaler {
+	return &v2beta1.HorizontalPodAutoscaler{
+		Spec: v2beta1.HorizontalPodAutoscalerSpec{
+			Metrics: []v2beta1.MetricSpec{
+				{
+					External: &v2beta1.ExternalMetricSource{
+						MetricName: metricName,
+						MetricSelector: &metav1.LabelSelector{
+							MatchLabels: labels,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func TestRemoveEntryFromConfigMap(t *testing.T) {
+	hpaCl := HPAWatcherClient{
+		clientSet: fake.NewSimpleClientset(),
+		ns:        "default",
+	}
+	_, err := hpaCl.clientSet.CoreV1().ConfigMaps("default").Get(datadogHPAConfigMap, metav1.GetOptions{})
+	fmt.Printf("err is %s \n", err)
+	assert.Contains(t, err.Error(), "configmaps \"datadog-hpa\" not found")
```

Comment:
Can we use this function here too https://github.com/kubernetes/apimachinery/blob/master/pkg/api/errors/errors.go#L418 instead of inspecting error string.

Reply:
💯